rust/connlib/tunnel/src/sockets.rs

```diff
@@ -10,39 +11,22 @@ use tokio::{io::Interest, net::UdpSocket};
 
 use crate::Result;
 
-pub struct Sockets {
+pub struct Sockets<S> {
```

Comment:
The reason this is generic is that on Linux, the Gateway doesn't protect its sockets, but the Clients do, so knowing the platform is not enough to know whether we should protect the sockets, right?

Reply:
I preferred reducing the platform-dependent code in connlib as it's harder to follow.

But that's a good point, on the gateway we could protect the sockets, shouldn't make much of a difference as long as no one adds a peer ip as a resource, but for that case it's worth calling protect. So I will update it.